### ISSUE TITLE
Only check for existence of executables if the run requires it

### ISSUE THREAD

**Issue by pansapiens** [MonashBioinformaticsPlatform/RNAsik-pipe#13]
Version: 1.4.8

RNAsik fails to run if bwa or hisat2 are missing, even if run won't use them based on the given commandline parameters.

eg, if `-align star` is specified, don't check for the existence of bwa or hisat2 executables unless they are actually required.

**Comment by serine** [comment]
yes, this is an issue. Sort of working on that issue, also put it on the road map for 1.4.10. However this is also a little bit related to @stu2 PR #6 should have it all cleaned by 1.4.10.

Thanks

**Comment by pansapiens** [comment]
A hacky workaround in the meantime is to specify, in `sik.config`:

```
hisat2Exe = false
bwaExe = false
```

This way it finds the `false` executable to keep it happy.

**Comment by serine** [comment]
that's not gonna fix it, problem is in `sikLog.bds` file [here](https://github.com/MonashBioinformaticsPlatform/RNAsik-pipe/blob/2c19a5dfd8c17238a75f036a821e5a16835bcbfc/src/sikLog.bds#L80) basically I simply take a log of all dependencies irrespective whether those tools are going to be used or not.. Like I said working on making this better [fix_logger branch](https://github.com/MonashBioinformaticsPlatform/RNAsik-pipe/tree/fix_logger)

this isn't a bad idea trying to use `false` there. I'll have to think about how its going to work, just not in this case

**Comment by pansapiens** [comment]
I'm aware of where the executable checks happen. Just a quick _workaround_ users can implement easily themselves if they happen to stumble upon this issue in the meantime vs. proper _fix_.

**Comment by serine** [comment]
Comment to myself, I started fix_logger branch, going to work there to fix these issues
